RandLAPACK/drivers/rl_hqrrp.hh

```diff
@@ -0,0 +1,873 @@
+                    );
+    } else if (typeid(T) == typeid(float)) {
+        LAPACK_slarfb( & side_, & trans_, & direction_, & storev_,  
+                    & m_, & n_, & k_, (float *) buff_U, & ldim_U, (float *) buff_T, & ldim_T, 
+                    (float *) buff_B, & ldim_B, (float *) buff_W, & ldim_W
+                    #ifdef LAPACK_FORTRAN_STRLEN_END
+                    , 1, 1, 1, 1
+                    #endif
+                    );
+    } else {
+        // Unsupported type
+    }
+    return;
+}
+
+// ============================================================================
+template <typename T>
+void _LAPACK_larf(
+    lapack::Side side,
+    int64_t m, int64_t n,
+    T *v, int64_t inc_v, T* tau,
+    T *C, int64_t ldc,
+    T *work
+){
+    char side_ = blas::side2char( side );
+    lapack_int inc_v_ = (lapack_int) inc_v;
+    lapack_int m_ = (lapack_int) m;
+    lapack_int n_ = (lapack_int) n;
+    lapack_int ldc_ = (lapack_int) ldc;
+    if (typeid(T) == typeid(double)) {
+        LAPACK_dlarf( & side_, & m_, & n_, 
+            (double *) v, & inc_v_,
+            (double *) tau,
+            (double *) C, & ldc_,
+            (double *) work
+            #ifdef LAPACK_FORTRAN_STRLEN_END
+            , 1
+            #endif
+            );
+    } else if (typeid(T) == typeid(float)) {
+        LAPACK_slarf( & side_, & m_, & n_, 
+            (float *) v, & inc_v_,
+            (float *) tau,
+            (float *) C, & ldc_,
+            (float *) work
+            #ifdef LAPACK_FORTRAN_STRLEN_END
+            , 1
+            #endif
+            );
+    } else {
+        // Unsupported type
+    }
+    return;
+}
+
+/* ========================================================================= */
+template <typename T>
+static T NoFLA_Normal_random_number( T mu, T sigma ) {
+    static int64_t     alternate_calls = 0;
+    static T  b1, b2;
+    T c1, c2, a, factor;
+
+    // Quick return.
+    if( alternate_calls == 1 ) {
+        alternate_calls = ! alternate_calls;
+        return( mu + sigma * b2 );
+    }
+    // Main loop.
+    do {
+        c1 = -1.0 + 2.0 * ( (T) rand() / RAND_MAX );
+        c2 = -1.0 + 2.0 * ( (T) rand() / RAND_MAX );
+        a = c1 * c1 + c2 * c2;
+    } while ( ( a == 0 )||( a >= 1 ) );
+    factor = sqrt( ( -2 * log( a ) ) / a );
+    b1 = c1 * factor;
+    b2 = c2 * factor;
+    alternate_calls = ! alternate_calls;
+    return( mu + sigma * b1 );
+}
+
+/* ========================================================================= */
+template <typename T>
+int64_t NoFLA_Normal_random_matrix( 
+    int64_t m_A, int64_t n_A, 
+    T * buff_A, int64_t ldim_A ) {
+//
+// It generates a random matrix with normal distribution.
+//
+    int64_t  i, j;
+
+    // Main loop.
+    #pragma omp parallel for
+    for ( j = 0; j < n_A; j++ ) {
+        for ( i = 0; i < m_A; i++ ) {
+            buff_A[ i + j * ldim_A ] = NoFLA_Normal_random_number( 0.0, 1.0 );
+        }
+    }
+
+    return 0;
+}
```

Comment:
Is this the only place that relies on ``NoFLA_Normal_random_number``? If so, then it should actually be really easy to have this depend only on RandBLAS for randomness.

Reply:
@rileyjmurray I think we already have this implemented in `fill_buff` with `DenseDist.family == DenseDistName::Gaussian`. Do you agree?

Reply:
We do have this implemented. But if this function isn't the only place in HQRRP that uses randomness then I wouldn't want to mix-and-match between the randomness in RandBLAS and in ``NoFLA_Normal_random_number``.

Reply:
Hell yeah. Then @TeachRaccooon please remove these two functions. Go to the place where you need to call ``NoFLA_Normal_random_matrix`` and replace it with appropriate RandBLAS calls. Also, have ``hqrrp`` accept a RandBLAS ``RNGState`` rather than an integer-valued random seed.

Reply:
Removed the two functions, replaced `NoFLA_Normal_random_matrix ` with a RandBLAS call, all works. 
For the point on `hqrrp` accepting random state as oppose to seed - this is an issue all across RandLAPACK. Also, we are using an older version of RandBLAS. 
This would be better off being included into the next PR not to stretch the scope of this one too much.
Idea approved by Riley per our Zoom meeting.